src/Generation/FieldDetails.php

```diff
@@ -94,12 +94,17 @@ class FieldDetails
     /** @var ?ResourceDetails The resource details, if this field is a resource; null otherwise. */
     public ?ResourceDetails $resourceDetails;
 
-    /** @var bool Whether tests and examples should use use a resource-type value. */
+    /** @var bool Whether tests and examples should use a resource-type value. */
     public bool $useResourceTestValue;
 
     /** @var ?int null if not in a one-of; otherwise the index of the one-of - ie every field in a oneof has the same index. */
     public ?int $oneOfIndex;
 
+    /**
+     * @var Vector *Readonly* Vector of FieldDetails. Contains all required subfields on this field, given the field
+     * is a message. Will be empty otherwise.
+     */
+    public Vector $requiredSubFields;
```

Comment:
> If so, should we refactor the generator to use this field? 

I think it'd be best to touch the gapic generator code when we go to do the work for updating the signatures.

> Will we want to update this to use google.api.method_annotation in the future, instead?

No, I don't think so. This is aso being used in this PR to help populate sample values for required fields of type `message` that may be nested in nature (i.e., not on the top level of the request).

> Also, does this handle OneOfs?

Yes, if the required annotation is applied.

> How does it handle when a field of a nested object is required?

If the subfield is a message, it will recursively process this same logic.